src/java.ts

```diff
@@ -269,12 +273,21 @@ async function getCachedFileLocation(md5: string, filename: string) {
   }
 }
 
-export async function fetchServerVersion(sonarHostUrl: string, token: string): Promise<SemVer> {
+export async function fetchServerVersion(
+  sonarHostUrl: string,
+  parameters: ScannerProperties,
+): Promise<SemVer> {
+  const token = parameters[ScannerProperty.SonarToken];
+  const proxyUrl = getProxyUrl(parameters);
+
   let version: SemVer | null = null;
   try {
     // Try and fetch the new version endpoint first
     log(LogLevel.DEBUG, `Fetching API V2 ${API_V2_VERSION_ENDPOINT}`);
-    const response = await fetch(token, { url: sonarHostUrl + API_V2_VERSION_ENDPOINT });
+    const response = await fetch(token, {
+      url: sonarHostUrl + API_V2_VERSION_ENDPOINT,
+      ...getHttpAgents(proxyUrl),
+    });
```

Comment:
if we use https://axios-http.com/docs/instance 

we can reduce all fetch requests to something like this
```suggestion
    const response = await fetch(API_V2_VERSION_ENDPOINT);
```

Reply:
yee although we'll still have to either pass around this instance 
```suggestion
    const response = await instance.fetch(API_V2_VERSION_ENDPOINT);
```

or have it defined globally at the module-level, I think this could be thought about in the task mentioned above (baby step? 😁)